util/httputil/compression.go

```diff
@@ -30,51 +31,32 @@ const (
 
 // Wrapper around http.Handler which adds suitable response compression based
 // on the client's Accept-Encoding headers.
-type compressedResponseWriter struct {
+type deflatedResponseWriter struct {
```

Comment:
Do we need this struct at all now?  `Write` is a passthrough, and `Close` might only be called in one place.

Reply:
We still need i guess, as this struct implements `http.ResponseWriter` and `zlib.NewWriter` does not implement all the methods needed.


What happens here is the all the methods needed by the `http.ResponseWriter` is implemented by the `deflatedResponseWriter` (via the `ResponseWriter`) and only the write and close method is "overridden" to use the `zlib.NewWriter` one.

Reply:
Make sense! changing it!